src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,495 @@
+####################################
+# Explicit Adaptive Adam-Bashforth solvers
+####################################
+# (Hairer & Wanner 1996, Vol I, p.398-400, 421-423)
+# ODE_MS Fixed-step, fixed-order multi-step numerical method
+#   with Adams-Bashforth-Moulton coefficients
+function ode_ms_adaptive(F, y0, tspan, order::Integer;reltol = 1.0e-5,
+                                                      abstol = 1.0e-8,
+                                                      minstep=abs(tspan[1] - tspan[1])/1e18,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
+                                                      facmin = .1,
+                                                      fac = .8)
+
+    # initialization of time span, which will have variable length
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = typeof(tspan[1])[]
+    push!(t,tstart)
+
+    # initialzation of y[n] and dy[n]
+    y = typeof(y0)[]
+    push!(y,y0)
+    dy = typeof(F(t[1], y[1]))[]
+
+    #initialization of step size array, while will have variable length as well
+    h = collect(initstep)
+    if h[1] == 0.
+        # initial guess at a step size
+        h[1], tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+    else
+        tdir = sign(tfinal - tstart)
+    end
+    #test with fixed step size
+    #@show h = diff(tspan)
+    #Note: h[n] = t[n+1]-t[n] => t[n+1] = t[n]+h[n]
+
+    #initialization of arrays used in calculation of further steps
+    #all initialized to a 2x2 array, which will need to expand to nxn
+    #over course of algorithm
+    c = zeros(3,3)
+    g = zeros(3,3)
+    b = zeros(2,2)
+    Φstar = Array{typeof(F(t[1], y[1])),2}(2,2)
+    Φ = Array{typeof(F(t[1], y[1])),2}(2,2)
+
+    LE = 1 #Error controller, defined here to be accessible throughout loop
+
+    #start at n=1, the first step
+    n = 1
+    while abs(t[n] - tfinal) > 0 && sign(tfinal-t[n])==tdir #&&  minstep < abs(h[n])
+        # Need to run the first several steps at reduced order
+        if n == 1
+            steporder = 1
+        else
+            steporder = min(n-1, order)
+        end
+        @show t[n]
+        #Current derivative: set dy[n] = F(t[n], y[n])
+        push!(dy, F(t[n], y[n]))
+
+        #allocate space for t[n+1], y[n+1], dy[n+1], h[n+1]
+        @show push!(t,t[1])
+        push!(y,y[1])
+        push!(dy,dy[1])
+        push!(h,h[1])
+
+        #resize coefficients arrays every power of two steporder
+        if log2(n)%1 == 0 && n>1
```

Comment:
This part of the code was actually unnecessary, so I don't do a check if `n` is a power of 2 anymore. Though, thanks for the link; there are some good tricks listed there.